src/lib.rs

```diff
@@ -1237,6 +1406,58 @@ pub async fn build(docker: &Docker, config: &Config) -> Result<(), Error> {
         }
     }
 
+    // Generate {appservice}.yml for appservices running on the host.
+    let generate_appservice_yml =
+        |appservice: &AppServiceConfig, host: &str, path: &Path| -> Result<(), Error> {
+            let parent = path
+                .parent()
+                .ok_or_else(|| anyhow!("Incorrect path {:?}, expected an absolute path", path))?;
+            std::fs::create_dir_all(parent)
+                .with_context(|| format!("Failed to create directory {:?}", parent))?;
+            let mut config = appservice.clone();
+            config.url.set_host(Some(host))?;
+            let writer = std::fs::File::create(&path)
+                .with_context(|| format!("Failed to create file {:?}", path))?;
+            serde_yaml::to_writer(writer, &config)
+                .with_context(|| format!("Failed to write file {:?}", path))?;
+            Ok(())
+        };
+
+    let guest_appservices_root = synapse_root.join("appservices");
+    let host_appservices_root = config.generated_user_files_dir().join("appservices");
```

Comment:
so you want people to tell their appservices to look for the registration in e.g. `/tmp/mx-tester/mjolnir/out/appservices/mjolnir.yml`?

Reply:
Yeah, that's the idea.
Open to suggestions on how to improve this!